searx/query.py

```diff
@@ -23,6 +23,7 @@
 )
```

Comment:
I know this file lack some comments.

A more simpler implementation (feel free to change it if you think it is better in another way, or it doesn't work).

Before https://github.com/asciimoo/searx/blob/99435381a84072b110c32004b2fb778af9b96f77/searx/query.py#L123

```python
            # external bang
            if len(query_part) > 2 and query_part[0:1] == '!!':
				self.external_bang = query_part[2:]
				parse_next = True
```

Remove _isExternalBang and _parseExternalBang from RawTextQuery,

In external_bang.py, replace ```_get_bang_query(query)``` by ```query.getSearchQuery()```. The good thing about is this
```!ddg :de searx``` will search for "searx" not for ":de searx".

Reply:
> In external_bang.py, replace _get_bang_query(query) by query.getSearchQuery(). The good thing about is this
!ddg :de searx will search for "searx" not for ":de searx".

I was unable to do this because the method getSearchQuery() does not exist in either query or the search variable. The rest I did implement and is working.

Reply:
Here a patch:
```patch
diff --git a/searx/external_bang.py b/searx/external_bang.py
index 098ae53f..7e1f48a6 100644
--- a/searx/external_bang.py
+++ b/searx/external_bang.py
@@ -16,42 +16,22 @@ def get_bang_url(search):
     """
 
     if search.external_bang:
-        query = str(search.query, "utf-8")
-        bang = _get_bang(search.external_bang)
-
-        if bang:
+        external_bang = _get_bang(search.external_bang)
+        if external_bang:
+            query = str(search.query, "utf-8")
             # TODO add region support.
-            bang_url = bang["regions"]["default"]
-
-            return bang_url.replace("{{{term}}}", _get_bang_query(query))
+            bang_url = external_bang["regions"]["default"]
+            return bang_url.replace("{{{term}}}", query)
     return None
 
 
-def _get_bang(user_bang):
+def _get_bang(external_bang):
     """
     Searches if the supplied user bang is available. Returns None if not found.
     :param user_bang: The parsed user bang. For example yt
     :return: Returns a dict with bangs data (check bangs_data.json for the structure)
     """
-    try:
-        return _get_bangs_data()[user_bang]
-    except KeyError:
-        return None
-
-
-def _get_bang_query(raw_query):
-    """
-    Gets the actual user search.
-    :param raw_query: The raw user query coming from the browser
-    :return: For example with the query &yt yes yes then this function will return yes yes.
-    """
-    slitted_raw_query = raw_query.split(" ")
-
-    full_query = ""
-    # Cause the first one is the bang like &yt
-    for raw_query in slitted_raw_query[1:]:
-        full_query += raw_query + "%20"
-    return full_query
+    return _get_bangs_data().get(external_bang)
 
 
 # Dont use this variable directly but access via _get_bangs_data
```

Reply:
I refactored its cleaner now. I thought the %20 was necessary but it was not I guess. Thanks for the patch! (and rest of the feedback)